pydantic/main.py

```diff
@@ -1337,7 +1333,7 @@ def _collect_bases_data(bases: tuple[type[Any], ...]) -> tuple[set[str], set[str
     class_vars: set[str] = set()
     private_attributes: dict[str, ModelPrivateAttr] = {}
     for base in bases:
-        if _base_class_defined and issubclass(base, BaseModel) and base != BaseModel:
+        if issubclass(base, BaseModel) and base != BaseModel:
```

Comment:
```suggestion
        if base == BaseModel:
            continue
```
(then dedent the remaining bits)